### DISCUSSION TITLE
If you're not specific which version to use of an action in a Github workflow, will it use the latest version?

### DISCUSSION THREAD

**Discussion by DanielGSoftware** [community/community#48058] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi everyone, I have a question regarding specifying a version to use when using a Github actions package. Let's assume I have the a simple workflow file that contains a step where I use an action.
```yml
    steps:
      - name: Checkout code
        uses: actions/checkout@v3
```

Here I define the version to be use to be `v3`. You can also put, for example, `v3.2.5`, which is the latest version.

My question is, if you just define the version as `v3`, which version is being used? 3.0.0 or the latest version 3.2.5?

Thanks in advance and sorry if this has already been asked. I tried to search it but couldn't find the answer.

**Comment by airtower-luna** [comment, marked as answer]
The answer is that it depends on the repository the action is in. The part after the `@` must be a valid Git ref (branch or tag name) or commit ID in the action repository. `actions/checkout@v3` will simply use whichever commit the `v3` tag or branch in the [`actions/checkout` repository](https://github.com/actions/checkout) points at.

If you look at the [list of tags for `actions/checkout`](https://github.com/actions/checkout/tags), you can see that currently (as I'm writing this) the `v3` and `v3.3.0` tags point at the same commit:

![grafik](https://user-images.githubusercontent.com/5814919/220154689-be45e5bf-bcbf-40f2-b96d-3cd4d3e59967.png)

But keep in mind: It's completely up to the maintainers of the action repository, there's no version parsing involved. :warning:

**Reply by DanielGSoftware** [reply]
Exactly what I was looking for, thank you very much!

**Reply by owaishussain227** [reply]
hii whts ur age

**Reply by dirkf** [reply]
So what happens if you don't specify any `@{version}`? Eg 
```yaml
    steps:
      - name: Checkout code
        uses: actions/checkout
```
You might guess that if this is valid it's equivalent to the deprecated `@{name_of_default_branch}` but doc is silent as to whether it's valid and what it means if so.
